src/fields/mod.rs

```diff
@@ -293,6 +249,101 @@ where
     std::iter::from_fn(move || Some(source()))
 }
 
+/// Precomputed values for the Lagrange basis polynomial-based implementation of `extend()`.
+pub struct LagrangeExtendContext<FE> {
+    nodes_len: usize,
+    evaluations: usize,
+    _phantom: PhantomData<FE>,
+}
+
+fn lagrange_extend_precompute<FE>(nodes_len: usize, evaluations: usize) -> LagrangeExtendContext<FE>
+where
+    FE: LagrangePolynomialFieldElement,
+{
+    LagrangeExtendContext {
+        nodes_len,
+        evaluations,
+        _phantom: PhantomData,
+    }
+}
+
+/// The extend method, as defined in [2.2.1][1] and [2.2.2][2]. We interpolate a polynomial of
+/// degree at most `nodes.len() - 1` from the provided evaluations at points `[0..nodes.len())`
+/// and then evaluate that polynomial at `[0, evaluations)`.
+///
+/// This implementation only works for large characteristic fields.
+///
+/// [1]: https://datatracker.ietf.org/doc/html/draft-google-cfrg-libzk-01#section-2.2.1
+/// [2]: https://datatracker.ietf.org/doc/html/draft-google-cfrg-libzk-01#section-2.2.2
+fn lagrange_extend<FE>(nodes: &[FE], context: &LagrangeExtendContext<FE>) -> Vec<FE>
+where
+    FE: LagrangePolynomialFieldElement,
+{
+    // For now we use the relatively straightforward method of computing Lagrange basis polynomials
+    // for each provided point.
+    //
+    // https://en.wikipedia.org/wiki/Lagrange_polynomial#Definition
+    assert_eq!(nodes.len(), context.nodes_len);
+    let evaluations = context.evaluations;
+    debug_assert!(
+        evaluations > nodes.len(),
+        "extend was given more evaluation points than were requested"
+    );
```

Comment:
Isn't this error message backwards? If `evaluations <= nodes.len()`, it means `extend` was asked to produce fewer evaluations than there are input nodes, not "more evaluation points than were requested".